jbcli/jbcli/cli/jb.py

```diff
@@ -364,6 +413,109 @@ def populate_env_with_secrets():
     return env
 
 
+@cli.command()
+@click.argument('env', nargs=1, required=False)
+@click.option('--noupdate', default=False,
+              help='Do not automatically update Docker image on start',
+              is_flag=True)
+@click.option('--noupgrade', default=False,
+              help='Do not automatically update jb and generators on start',
+              is_flag=True)
+@click.pass_context
+def freshstart(ctx, env, noupdate, noupgrade):
+    """Configure the environment and start Juicebox
+    """
+    if dockerutil.is_running():
+        echo_warning('An instance of Juicebox is already running')
+        echo_warning('Run `jb stop` to stop this instance.')
+        return
+
+    path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
+    os.chdir(path)
+    if env is None:
+        click.echo("Welcome to devlandia!")
+        click.echo('Gathering data on available environments...\n')
+        # get a list of all the current images
+        tags = dockerutil.image_list(showall=True, print_flag=False, semantic=False)
+        # these are the tags we want to look for in the list of images
+
+        # Generate suffixes for each environment that contain information on what image is
+        # associated with a tag, and when those images were published
+        extra_lookup = {}
+
+        jb_select = stash.get('jb_select', {})
+        if not isinstance(jb_select, dict):
+            jb_select = {
+                'test': jb_select
+            }
+
+        for env, selected_tag in jb_select.items():
+            extra_lookup[env] = selected_tag
+
+        for row in tags:
+            tag, pushed, human_readable, tag_priority, is_semantic_tag, stable_version = row
+
+            if tag == 'master':
+                extra_lookup['master'] = 'stable/master - ({}) published {}'.format(stable_version, human_readable)
+            if tag == 'develop':
+                extra_lookup['dev'] = 'dev - (develop) published {}'.format(human_readable)
+                extra_lookup['core'] = 'core - (develop) published {}'.format(human_readable)
+            for env, selected_tag in jb_select.items():
+                if tag == selected_tag:
+                    extra_lookup[env] = '{} - ({}, set with `jb select`) published {}'.format(env, selected_tag, human_readable)
+
+        if 'test' not in extra_lookup:
+            extra_lookup['test'] = 'test - set with `jb select`'
+
+        env_choices = [
+            {'name': extra_lookup.get('master', 'master'), 'value': 'stable'},
```

Comment:
this could be generated based on the contents of the `environments` directory, but no big deal